25.0 release notes: batch 3

317brian opened the pull request:
Replaces https://github.com/apache/druid/pull/13357/

This takes us up to November 23 (after the 25.0.0 branch was created)

kfaraz commented on docs/next-release-notes.md:
```diff
@@ -0,0 +1,496 @@
+
+https://github.com/apache/druid/pull/13367
+
+### Segment balancing
```
Do you think it would be better to club the `segment balancing`, `segment replication` and `segment assignment` under one heading? They are all related coordinator-side changes which improve segment loading and balancing.

317brian replied:
I'll move them under an H3 called `Segments` and put assignment, balancing, replication, and discovery as H4s underneeth

kfaraz commented on docs/next-release-notes.md:
```diff
@@ -0,0 +1,496 @@
+
+https://github.com/apache/druid/pull/13144
+
+### 
```
empty heading?

kfaraz commented on docs/next-release-notes.md:
```diff
@@ -0,0 +1,496 @@
+
+https://github.com/apache/druid/pull/13311
+
+### cachingCost balancer strategy
```
This too could be an item inside `segment loading and balancing improvements`

2bethere commented on docs/next-release-notes.md:
```diff
@@ -0,0 +1,517 @@
+
+https://github.com/apache/druid/pull/13238 
+
+### Idle configs for the Supervisor
```
I think we should explain to the reader that it's now possible to set supervisor to ideal. This is useful in cases of freeing up slots such that auto-scaling can kick in more effectively.

2bethere commented on docs/next-release-notes.md:
```diff
@@ -0,0 +1,517 @@
+
+https://github.com/apache/druid/pull/13321
+
+### New metrics for streaming ingestion
+
```
Those should go into metrics section.

2bethere commented on docs/next-release-notes.md:
```diff
@@ -0,0 +1,517 @@
+https://github.com/apache/druid/pull/12102
+
+### taskActionType dimension for task/action/run/time metric
+
```
This should also go into the metrics section.

2bethere commented on docs/next-release-notes.md:
```diff
@@ -0,0 +1,517 @@
+
+https://github.com/apache/druid/pull/13367
+
+#### Segment batch balancing
```
This needs to be highlighted as a behavior change after the highlight section.

kfaraz replied:
```suggestion
#### Segment batch sampling
```

2bethere commented on docs/next-release-notes.md:
```diff
@@ -0,0 +1,517 @@
+
+https://github.com/apache/druid/pull/13391
+
+#### Segment discovery
```
This is also a behavior change that we should highlight.

2bethere replied:
As part of this highlight, you should include details on how to revert to old behavior.

2bethere commented on docs/next-release-notes.md:
```diff
@@ -0,0 +1,517 @@
+
+https://github.com/apache/druid/pull/13178
+
+### Docker improvements
```
This should go into the "Dependency updates" section

2bethere commented on docs/next-release-notes.md:
```diff
@@ -0,0 +1,517 @@
+
+
+
+### Kill tasks do not include markAsUnuseddone
```
This is also a behavior change.

2bethere commented on docs/next-release-notes.md:
```diff
@@ -0,0 +1,517 @@
+
+https://github.com/apache/druid/pull/13059
+
+### Improved supervisor termination
```
This should be moved together with other supervisor idle stuff.

2bethere commented on docs/next-release-notes.md:
```diff
@@ -0,0 +1,517 @@
+
+https://github.com/apache/druid/pull/13064
+
+### Improved metric reporting
```
This should go into metrics section.

2bethere commented on docs/next-release-notes.md:
```diff
@@ -0,0 +1,517 @@
+
+https://github.com/apache/druid/pull/13262
+
+## Web console
```
Web console should be moved above extensions

2bethere commented on docs/next-release-notes.md:
```diff
@@ -0,0 +1,517 @@
+
+https://github.com/apache/druid/pull/13291
+
+### Query history
```
This should move to MSQ section.

2bethere reviewed (commented):
Made some comments, mostly about shuffling content around. Thanks for pulling this together. 

I think the new format with links are great!

kfaraz commented on docs/next-release-notes.md:
```diff
@@ -0,0 +1,517 @@
+
+Batch sampling is now the default method for sampling segments during balancing as it performs significantly better than the alternative when there is a large number of used segments in the cluster.
+
+As part of this change, the following properties have been deprecated and will be removed in future releases:
+
+- `coordinator dynamic config useBatchedSegmentSampler`
+- `coordinator dynamic config percentOfSegmentsToConsiderPerMove`
+- non-batch method of sampling segments used during coordinator duty `BalanceSegments`
```
```suggestion
As part of this change, the following have been deprecated and will be removed in future releases:

- coordinator dynamic config `useBatchedSegmentSampler`
- coordinator dynamic config `percentOfSegmentsToConsiderPerMove`
- non-batch method of sampling segments used by coordinator duty `BalanceSegments`
```

kfaraz commented on docs/next-release-notes.md:
```diff
@@ -0,0 +1,517 @@
+- `coordinator dynamic config percentOfSegmentsToConsiderPerMove`
+- non-batch method of sampling segments used during coordinator duty `BalanceSegments`
+
+The unused coordinator property `druid.coordinator.loadqueuepeon.repeatDelay` has been removed.
+
+Use only `druid.coordinator.loadqueuepeon.http.repeatDelay` to configure repeat delay for the HTTP-based segment loading queue.
```
These two items should either be under the heading `Segment assignment` or a separate heading titled `Segment load queue peon`.

kfaraz commented on docs/next-release-notes.md:
```diff
@@ -0,0 +1,517 @@
+
+https://github.com/apache/druid/pull/13356
+
+### Segmen loading and balancings
```
```suggestion
### Segment loading and balancing
```

317brian commented on docs/next-release-notes.md:
```diff
@@ -0,0 +1,517 @@
+This release includes several improvements for querying:
+
+* Exposed HTTP response headers for SQL queries (https://github.com/apache/druid/pull/13052)
+* Added the `shouldFinalize` feature for HLL and quantiles sketches. Druid will no longer finalize aggregators when:
```
@vtlim will work on the doc updates

vtlim replied:
PR: https://github.com/apache/druid/pull/13524

317brian commented:
@2bethere the latest commit addresses the comments from your review.

@kfaraz there's 1 more commit planned to capture the PRs from the last milestone. After that, we should be done once it's reviewed.

317brian commented:
[`ad84523` (#13429)](https://github.com/apache/druid/pull/13429/commits/ad845238ee1fe45baa0afe901da015bc9c3a8a6f) contains what should be the last batch from `git log` and milestone 48. 

The only one I haven't added is https://github.com/apache/druid/pull/13365 from the milestone since the original PR is still open.

cryptoe commented on docs/next-release-notes.md:
```diff
@@ -0,0 +1,618 @@
+
+## Multi-stage query task engine
+
+### Worker retry
+
+The MSQ controller task now retries worker tasks if they fail up to two times. The maximum number of total retries for a job across all workers is 30.
+
+https://github.com/apache/druid/pull/13353
+
```
This might not make it to the 25.0 release. So I suggest removing this. 
```suggestion
```

317brian commented:
@kfaraz I've added the start-druid script entry to the release notes, so we should be all caught up.

kfaraz commented on docs/next-release-notes.md:
```diff
@@ -21,19 +21,88 @@ title: "WIP release notes for 25.0"
+| `druid.coordinator.loadqueuepeon.type` for segment management | http | curator |
+| `druid.indexer.runner.type` for the Overlord | httpRemote | local |
+
+To use ZooKeeper instead of HTTP, change the values for the properties back to the previous defaults.
```
We should also mention that the Zookeeper based implementations for the above items are now deprecated and will be removed in subsequent releases.

kfaraz commented on docs/next-release-notes.md:
```diff
@@ -119,21 +190,84 @@ https://github.com/apache/druid/pull/13133
+Metrics for batch segment allocation (dims: dataSource, taskActionType=segmentAllocate):
+
+- `task/action/batch/runTime`: Milliseconds taken to execute a batch of task actions. Currently only being emitted for [batched `segmentAllocate` actions
+- task/action/batch/queueTime: Milliseconds spent by a batch of task actions in queue. Currently only being emitted for [batched `segmentAllocate` actions](#segment-batch-allocation).
```
```suggestion
- `task/action/batch/queueTime`: Milliseconds spent by a batch of task actions in queue. Currently only being emitted for [batched `segmentAllocate` actions](#segment-batch-allocation).
```

kfaraz commented on docs/next-release-notes.md:
```diff
@@ -119,21 +190,84 @@ https://github.com/apache/druid/pull/13133
+
+Segment allocation can now be performed in batches, which can improve performance and decrease ingestion lag. For more information about this change, see [Segment batch allocation](#segment-batch-allocation). The following metrics correspond to those changes.
+
+Metrics for batch segment allocation (dims: dataSource, taskActionType=segmentAllocate):
```
```suggestion
Metrics emitted for a batch segment allocate requests (dims: dataSource, taskActionType=segmentAllocate):
```

kfaraz commented on docs/next-release-notes.md:
```diff
@@ -119,21 +190,84 @@ https://github.com/apache/druid/pull/13133
+- `task/action/batch/size`: Number of task actions in a batch that was executed during the emission period. Currently only being emitted for [batched `segmentAllocate` actions](#segment-batch-allocation).
+- `task/action/batch/attempts`: Number of execution attempts for a single batch of task actions. Currently only being emitted for [batched `segmentAllocate` actions](#segment-batch-allocation). 
+
+Metrics for a request (dims: taskId, taskType, dataSource, taskActionType=segmentAllocate):
```
```suggestion
Metrics emitted for a single segment allocate request (dims: taskId, taskType, dataSource, taskActionType=segmentAllocate):
```

kfaraz commented on docs/next-release-notes.md:
```diff
@@ -239,6 +362,10 @@ https://github.com/apache/druid/pull/12177
 
 ## Operations
 
+### Deploying Druid
```
This item should be in the highlights.

kfaraz commented on docs/next-release-notes.md:
```diff
@@ -239,6 +362,10 @@ https://github.com/apache/druid/pull/12177
 
+### Deploying Druid
+
+There is now a Python installation script available for Druid that simplifies deployments that don't fit the parameters of the singel-server profiles, such as the nano or micro-quickstart profiles. This `start-druid` script lets you override Druid settings to fit your needs. For more information, see [Single server deployment](https://druid.apache.org/docs/latest/operations/single-server.html).
```
The new script simplifies deployments for all cases, not just the ones that don't fit the existing single-server profiles. We could probably rephrase this section as follows:

```suggestion
The new `start-druid` script greatly simplifies starting any combination of Druid services on a single-server. It comes pre-packaged with the required configs and can be used to launch a fully functional Druid cluster simply by invoking `./start-druid`. For the experienced Druids, it also gives you complete control over the runtime properties and JVM arguments to have a cluster that exactly fits your needs.

The `start-druid` script deprecates the existing profiles such as `start-micro-quickstart` and `start-nano-quickstart`. These profiles may be removed in future releases. For more information, see [Single server deployment](https://druid.apache.org/docs/latest/operations/single-server.html).
```

@vogievetsky , does this capture our sentiments adequately?

vtlim replied:
I [moved this section to the highlights](https://github.com/apache/druid/pull/13429#discussion_r1050809228) and also used the text from this comment.

kfaraz commented:
Closing this as Druid 25.0 is released.
